lang/java/reef-examples/src/main/java/org/apache/reef/examples/data/loading/DataLoadingREEFOnSpark.java

```diff
@@ -0,0 +1,159 @@
+ */
+import org.apache.hadoop.mapred.TextInputFormat;
+import org.apache.reef.annotations.audience.ClientSide;
+import org.apache.reef.client.DriverConfiguration;
+import org.apache.reef.driver.evaluator.EvaluatorRequest;
+import org.apache.reef.runtime.spark.job.SparkDataLoadingRequestBuilder;
+import org.apache.reef.runtime.local.client.LocalRuntimeConfiguration;
+import org.apache.reef.runtime.yarn.client.YarnClientConfiguration;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Injector;
+import org.apache.reef.tang.JavaConfigurationBuilder;
+import org.apache.reef.tang.Tang;
+import org.apache.reef.tang.annotations.Name;
+import org.apache.reef.tang.annotations.NamedParameter;
+import org.apache.reef.tang.exceptions.BindException;
+import org.apache.reef.tang.exceptions.InjectionException;
+import org.apache.reef.tang.formats.CommandLine;
+import org.apache.reef.util.EnvironmentUtils;
+import org.apache.reef.runtime.spark.job.SparkRunner;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Client for the data loading demo app running on spark executors.
+ */
+@ClientSide
+public final class DataLoadingREEFOnSpark {
+
+  private static final Logger LOG = Logger.getLogger(DataLoadingREEFOnSpark.class.getName());
+
+  /**
+   * The upper limit on the number of Evaluators that the local resourcemanager will hand out concurrently.
+   */
+  private static final int MAX_NUMBER_OF_EVALUATORS = 16;
+
+  private static final int NUM_SPLITS = 6;
+  private static final int NUM_COMPUTE_EVALUATORS = 2;
+
+
+  private String inputPath;
+
+
+  public static void main(final String[] args)
+      throws InjectionException, BindException, IOException {
+
+    final Tang tang = Tang.Factory.getTang();
+
+
+    final JavaConfigurationBuilder cb = tang.newConfigurationBuilder();
+
+    new CommandLine(cb)
+        .registerShortNameOfClass(DataLoadingREEFOnSpark.Local.class)
+        .registerShortNameOfClass(DataLoadingREEFOnSpark.TimeOut.class)
+        .registerShortNameOfClass(DataLoadingREEFOnSpark.InputDir.class)
+        .processCommandLine(args);
+
+    // TODO[REEF-1791] remove hardcoding and figure out where to put these files
+    String inputPath="file:///"+System.getenv("REEF_HOME") +"/pom.xml";
+
+    final Injector injector = tang.newInjector(cb.build());
+
+    final boolean isLocal = injector.getNamedInstance(DataLoadingREEFOnSpark.Local.class);
+
+
+
+    final Configuration runtimeConfiguration;
+    if (isLocal) {
+      LOG.log(Level.INFO, "Running Data Loading reef on spark demo on the local runtime");
+      runtimeConfiguration = LocalRuntimeConfiguration.CONF
+          .set(LocalRuntimeConfiguration.MAX_NUMBER_OF_EVALUATORS, MAX_NUMBER_OF_EVALUATORS)
+          .build();
+    } else {
+      LOG.log(Level.INFO, "Running Data Loading reef on spark demo on YARN");
+      runtimeConfiguration = YarnClientConfiguration.CONF.build();
+    }
```

Comment:
I don't think we need either Local or Yarn runtime configuration. REEF should define a new `SparkRuntimeConfiguration` that piggybacks on Spark framework and itself is agnostic of whether the underlying Spark runs locally or on YARN.

For now, this new Spark runtime config can have very few parameters (if any), as we can implement most of the stuff as part of the Spark data loader.

Reply:
Ok I was trying to piggyback the yarn runtime config, will create a SparkRuntimeConfiguration that abstracts how spark runs underneath